### PR TITLE
feat: add test for the user card component

### LINKED ISSUE #1108: Create src/components/cards/Wallet component test

### LINKED ISSUE #1109: Create src/components/cards/User component test

### PR DESCRIPTION
**Pull request by mhenriette**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [] Documentation has been updated to reflect your changes.
- [x] Tests have been added or updated to reflect your changes.
- [x] All tests pass.

---

## Pull Request Details
Add tests for user card component

## Breaking Changes

none

## Issues Fixed
#1109

### CODE REVIEW

File: __tests__/components/cards/User.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    });
+  });
+
+  test("Renders the correct tag and currecy when teamMembers is not empty", () => {
+    expect(screen.getByTestId("currencyId")).toBeInTheDocument();
+    expect(screen.getByTestId("tag")).toBeInTheDocument();
+  });
+
+  test("Calls onClock prop when clicked", () => {
```
Comment (Jonath-z) [inline __tests__/components/cards/User.test.tsx:54]:
```suggestion
  test("Calls onClick prop when clicked", () => {
```

File: __tests__/components/cards/User.test.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  useRouter: jest.fn(),
+}));
+const teamMembers = [mockUser];
+const handleClick = jest.fn(() => {});
+const mockPush = jest.fn();
+const mockUseRouter = {
+  locale: "en",
+  push: mockPush,
+};
+(useRouter as jest.Mock).mockReturnValue(mockUseRouter);
```
Comment (Samantha-KY) [inline __tests__/components/cards/User.test.tsx:18]: I don't think you need this mock as you mocked useRouter on line 8

File: __tests__/components/cards/User.test.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        <div>test link</div>
+      </UserCard>
+    );
+  });
+  // Mock window.getSelection
+  window.getSelection = jest.fn().mockReturnValue({
+    type: "Caret",
+  });
+  test("Renders the card component", () => {
+    const teamMembers = [mockUser];
```
Comment (Samantha-KY) [inline __tests__/components/cards/User.test.tsx:37]: There is no need to repeat this as you declared it on top line 11

File: __tests__/components/cards/User.test.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const linkElements = screen.getAllByRole("link");
+    linkElements.forEach((element, index) => {
+      if (teamMembers[index]) {
+        const expectedHref = `/profile/${teamMembers[index].username}`;
+        expect(element).toHaveAttribute("href", expectedHref);
+      }
+    });
+  });
+
+  test("Renders the correct tag and currecy when teamMembers is not empty", () => {
```
Comment (Samantha-KY) [inline __tests__/components/cards/User.test.tsx:49]: we are using the keyword "it" for the tests, please keep the consistency

**Review by Jonath-z — APPROVED** [review]

File: __tests__/components/cards/User.test.tsx (hunk 2 above)
Comment (mhenriette) [reply __tests__/components/cards/User.test.tsx]: we need to mock the returned value as well

**Review by Samantha-KY — APPROVED** [review]

**Pull request merged**